### ISSUE TITLE
Support for attachments inside the .zip file

### ISSUE THREAD

**Issue by santi-ifontana** [4350pChris/slack-vuesualizer#838]
Hello. I love your app, so far it is the nicest and fastest solution for visualizing Slack export files.

But quick question, is it possible or planned to support attachments in the .zip?
I have used this tool: https://github.com/grundleborg/slack-advanced-exporter?tab=readme-ov-file
Why? Because we are going to close our Slack account and we need to keep a working archival for legal purposes (including the attachments).

Please, would love to hear from you.

Thank you.

**Comment by 4350pChris** [comment]
Could you maybe produce a sanitized .zip file with attachments for me to test? That would make it a lot easier, otherwise I don't even really know what I'm dealing with.
I'll try to look it up though, if you provide a file I'm fairly sure I can do that in a timely manner.

**Comment by santi-ifontana** [comment]
Hey Chris,

Here is the new .zip file with the attachments inside:
<img width="1156" alt="Screenshot 2024-06-28 at 14 08 35" src="https://github.com/4350pChris/slack-vuesualizer/assets/165914151/fde46014-ca82-4ee7-8c88-629f1430944a">
<img width="1203" alt="Screenshot 2024-06-28 at 14 08 42" src="https://github.com/4350pChris/slack-vuesualizer/assets/165914151/12f6380d-93c7-4cb5-a521-e78f5d5269ec">
<img width="1193" alt="Screenshot 2024-06-28 at 14 08 49" src="https://github.com/4350pChris/slack-vuesualizer/assets/165914151/9372f279-93f6-44b3-9e15-02728d91b07a">

Checking a channel .json file, the links are kept unchanged.
`                "url_private": "https:\/\/files.slack.com\/files-pri\/T08KDGR8S-F01QF9TSN9P\/image_from_ios.png?t=xoxe-<token>",
                "url_private_download": "https:\/\/files.slack.com\/files-pri\/T08KDGR8S-F01QF9TSN9P\/download\/image_from_ios.png?t=xoxe-<token>",
                `
             
As the .zip is over 100 GB I cannot even finish unzipping it before I run out of space.
Let me know if this helps you or if you really need the .zip, I will check how I can get it unzipped and sanitized.

Just to mention, the tool used to feed the attachment into the .zip is part of the mattermost import process: https://docs.mattermost.com/onboard/migrate-from-slack.html

Thanks.

**Comment by 4350pChris** [comment]
So what exactly would you expect from this app? I don't quite understand what to do with the attachments. Make them downloadable in the file section instead of having the URL to the download?

**Comment by santi-ifontana** [comment]
Hey,

Would be great to be able to have the files downloaded locally and not dependent on the Slack servers.
In our case, we are planning on ending our Slack subscription and we would like to keep an archival running.

This solution, although it crashes often, it allows you to "cache" all the files: https://github.com/thayakawa-gh/SlackLogViewer

So, to sum up, instead of trying to upload a 113 GB .zip file with all the attachments (with the tool I provided earlier), maybe upload the regular .zip archive offered by the official Slack export tool and then, have the option to download locally all the files.

Sorry, English is not my first language. I hope I am clear enough.

Thank you!

**Comment by 4350pChris** [comment]
Ah I see. So basically just a bulk download of files? I think I can manage that, although I can't test it with as many attachments. Maybe chinking would be a good idea here, so you don't have to download the whole >100GB at once?

**Comment by santi-ifontana** [comment]
> So basically just a bulk download of files?
Yes, so the app can run standalone (no need to reach Slack's servers).

>  I can't test it with as many attachments
We can help you test.

> Maybe chunking would be a good idea here, so you don't have to download the whole >100GB at once?
That is no problem. We have a small server running in GCP, so if it downloads all at once or in chunks is fine.

What would be great is to be able to define the mount dir in Docker / Composer, so I can manage where the files are stored.

Thank you.
